src/memory_provider.c

```diff
@@ -24,11 +25,70 @@ typedef struct umf_memory_provider_t {
     void *provider_priv;
 } umf_memory_provider_t;
 
+static umf_result_t umfDefaultPurgeLazy(void *provider, void *ptr,
+                                        size_t size) {
+    (void)provider;
+    (void)ptr;
+    (void)size;
+    return UMF_RESULT_ERROR_NOT_SUPPORTED;
+}
+
+static umf_result_t umfDefaultPurgeForce(void *provider, void *ptr,
+                                         size_t size) {
+    (void)provider;
+    (void)ptr;
+    (void)size;
+    return UMF_RESULT_ERROR_NOT_SUPPORTED;
+}
+
+static umf_result_t umfDefaultAllocationSplit(void *provider, void *ptr,
+                                              size_t totalSize,
+                                              size_t firstSize) {
+    (void)provider;
+    (void)ptr;
+    (void)totalSize;
+    (void)firstSize;
+    return UMF_RESULT_ERROR_NOT_SUPPORTED;
+}
+
+static umf_result_t umfDefaultAllocationMerge(void *provider, void *lowPtr,
+                                              void *highPtr, size_t totalSize) {
+    (void)provider;
+    (void)lowPtr;
+    (void)highPtr;
+    (void)totalSize;
+    return UMF_RESULT_ERROR_NOT_SUPPORTED;
+}
+
+void assignOpsExtDefaults(umf_memory_provider_ops_t *ops) {
+    if (!ops->ext.purge_lazy) {
+        ops->ext.purge_lazy = umfDefaultPurgeLazy;
+    }
+    if (!ops->ext.purge_force) {
+        ops->ext.purge_force = umfDefaultPurgeForce;
+    }
+    if (!ops->ext.allocation_split) {
```

Comment:
So, can we check it here that either both `allocation_split` and `allocation_merge` is set or neither is? Also, we could mention that in the documentation.

Reply:
Done. I extended the validation function to check that both are set or both are NULL. Also, I extended corresponding comments in the `umf_memory_provider_ext_ops_t` structure.